### PR TITLE
Widget Editor: Refactor header using the DocumentTools component from the editor package

### PR DESCRIPTION
**Pull request by t-hamano**
Related to #57214

## What?

This PR refactors the header in the Widget Editor using the `DocumentTools` component from the `@wordpress/editor` package instead of the proprietary `DocumentTools` component.

At the same time, we want the "Show button text labels" setting to be applied to the header area as well.

### Before

Document tool buttons are 36px in size unlike other editors. Additionally, the focus outline is cut off.

image

Mobile Layout:

image

### After

The document tools button size is 32 pixels, similar to other editors. Also, the focus outline is not cut.

image

The Widget Editor doesn't have the preferences modal, but when you turn on "Show button text labels" in the Post Editor or the Site Editor, the buttons in the header display text instead of icons.

image

Mobile Layout: 

image

image

## Why?

The Widget Editor header has a different UI compared to other editors. The approach taken in #57214 to unify `DocumentTools` component should also be applicable to the Widget Editor.

## How?

- Replace `DocumentTools`
- Reflect the "Show button text labels" setting
- Added deprecation to `setIsInserterOpened` and `setIsListViewOpened` actions
- Adjusting CSS to achieve the same layout as other editors

## Testing Instructions

- Inserting blocks and list views should work as before.
- Edit/Select mode and Undo/Redo should work as before.
- With a _closed_ widget area selected, clicking on the main block inserter should open that widget area.
- When you enable the Top Toolbar and narrow the browser width, a horizontal scrollbar should appear on the block toolbar.

### CODE REVIEW

File: packages/edit-widgets/src/components/layout/interface.js [hunk 1]
```diff
@@ -82,6 +99,21 @@ function Interface( { blockEditorSettings } ) {
...
+			// Open the last selected widget area when opening the inserter.
+			setIsWidgetAreaOpen( widgetAreaClientId, true );
+		}
+	}, [
+		widgetAreaClientId,
+		isInserterOpened,
+		isLastSelectedWidgetAreaOpen,
+		selectBlock,
+		setIsWidgetAreaOpen,
+	] );
```
Comment (t-hamano) [inline packages/edit-widgets/src/components/layout/interface.js:115]: The unique processing that existed in the `DocumentTools` component of the Widget Editor has been moved here. If the selected widget area is closed when you click the inserter button, the widget area will be opened.

a5a85f0d3fc3ea443426f1891966f99d

**Comment by talldan** [comment]
IIRC there was a reason `@wordpress/editor` wasn't used in widgets editors. Or maybe not and it wasn't needed at the time.

@adamziel or @noisysocks might know more.

**Comment by youknowriad** [comment]
I also checked whether the widgets editor should use the editor package or not and decided that for now it should not.

The editor package components rely on the fact that the editor is about editing a given entity and this is provided as a prop to the EditorProvider component which is responsible for initializing the store properly.

As it stands, this abstraction is not applicable to the widgets editor because it has no root entity to be edited.

While it's true that some components can work like shown here, this is only due to the fact that we have global stores in WP but this is not a given.

So I think we should not try to reuse the package and the components for the moment.

---

That said if we think about it, the widget areas are not different than template parts conceptually, so in a longer timeframe, I see no reason for not moving these to the site editor and just editing them exactly like template parts

**Comment by t-hamano** [comment]
@talldan @youknowriad  Thank you for the advice. Now, I would like to close this PR and consider adjusting the style while keeping DocumentTools in the Widget Editor 👍

**Pull request closed without merging**